### PR TITLE
Random by channel

### PR DESCRIPTION
**Pull request by vanillaHafer**
Provides the ability to find random posts by a specific channel

# Channel page

image

### CODE REVIEW

**Review by vnegrisolo — COMMENTED** [review]
very nice PR, thanks for that, here're are a few personal feedback on that:

- I'd move this new action to the channel controller;
- I'd try to reuse the same "show.html" template from the channel controller but with a single post, you may have to change the view helper that generates the header to instead of showing a count of posts, you show a "random posts about #rails";
- I'd reuse the dice from the bottom left menu "random" as the link that activates this feature;

**Comment by vanillaHafer** [comment]
@vnegrisolo Thanks for the great feedback.

I went ahead and implemented those changes you suggested. Feel free to let me know if there's anything else you would do differently.

File: lib/tilex_web/controllers/channel_controller.ex [hunk 1]
```diff
@@ -17,7 +17,27 @@ defmodule TilexWeb.ChannelController do
...
+      page: page,
+      random: false
+    )
+  end
+
+  def random_by_channel(conn, %{"channel" => channel_name} = params) do
+    page =
+      params
+      |> Map.get("page", "1")
+      |> String.to_integer()
```
Comment (avogel3) [inline lib/tilex_web/controllers/channel_controller.ex:29]: Think this variable is unused?

**Comment by avogel3** [comment]
Nice work, @vanillaHafer ⭐

File: lib/tilex_web/controllers/channel_controller.ex (hunk 1 above)
Comment (vanillaHafer) [reply lib/tilex_web/controllers/channel_controller.ex]: My bad, overlooked that. Meant to remove it. Took care of it!

**Pull request merged**